internal/clientconn/conn.go

```diff
@@ -86,18 +89,12 @@ func newConn(opts *newConnOpts) (*conn, error) {
 		}
 	}
 
-	handlerOpts := &pg.NewOpts{
-		PgPool:    opts.pgPool,
-		L:         l,
-		PeerAddr:  peerAddr,
-		Metrics:   opts.handlersMetrics,
-		StartTime: opts.startTime,
-	}
 	return &conn{
 		netConn: opts.netConn,
 		mode:    opts.mode,
 		l:       l.Sugar(),
-		h:       pg.New(handlerOpts),
+		h:       opts.handler,
+		m:       NewConnMetrics(),
```

Comment:
That breaks metrics gathering since we only register a listener's collector in Prometheus registry, not all conneciton (and we can't register each connection individually).

That should be a shared instance passed via opts from listener.

Reply:
moved to listerens' metrics.